jerry-core/jerry-snapshot.c

```diff
@@ -630,3 +632,430 @@ jerry_exec_snapshot (const void *snapshot_p, /**< snapshot */
+                                               buffer_end_p,
+                                               (const char *) str_buffer_p,
+                                               str_buffer_size);
+
+  ECMA_FINALIZE_UTF8_STRING (str_buffer_p, str_buffer_size);
+
+  return new_buffer_p;
+} /* jerry_append_ecma_string_to_buffer */
+
+/**
+ * Append an unsigned number to the specified buffer.
+ *
+ * @return the position of the buffer pointer after copy.
+ */
+static uint8_t *
+jerry_append_number_to_buffer (uint8_t *buffer_p, /**< buffer */
+                               uint8_t *buffer_end_p, /**< the end of the buffer */
+                               lit_utf8_size_t number) /**< number */
+{
+  lit_utf8_byte_t uint32_to_str_buffer[ECMA_MAX_CHARS_IN_STRINGIFIED_UINT32];
+  lit_utf8_size_t utf8_str_size = ecma_uint32_to_utf8_string (number,
+                                                              uint32_to_str_buffer,
+                                                              ECMA_MAX_CHARS_IN_STRINGIFIED_UINT32);
+
+  JERRY_ASSERT (utf8_str_size <= ECMA_MAX_CHARS_IN_STRINGIFIED_UINT32);
+
+  return jerry_append_chars_to_buffer (buffer_p,
+                                       buffer_end_p,
+                                       (const char *) uint32_to_str_buffer,
+                                       utf8_str_size);
+} /* jerry_append_number_to_buffer */
+
+/**
+ * Check whether the passed ecma-string is a valid identifier.
+ *
+ * @return true, if the ecma-string is a valid identifier,
+ *         false - otherwise.
+ */
+static bool
+ecma_string_is_valid_identifier (ecma_string_t *string_p)
+{
+  bool result = false;
+
+  ECMA_STRING_TO_UTF8_STRING (string_p, str_buffer_p, str_buffer_size);
+
+  if (lit_char_is_identifier_start (str_buffer_p))
+  {
+    const uint8_t *str_start_p = str_buffer_p;
+    const uint8_t *str_end_p = str_buffer_p + str_buffer_size;
+
+    result = true;
+
+    while (str_start_p < str_end_p)
+    {
+      if (!lit_char_is_identifier_part (str_start_p))
+      {
+        result = false;
+        break;
+      }
+      lit_utf8_incr (&str_start_p);
+    }
+  }
+
+  ECMA_FINALIZE_UTF8_STRING (str_buffer_p, str_buffer_size);
+
+  return result;
+} /* ecma_string_is_valid_identifier */
+
+#endif /* JERRY_ENABLE_SNAPSHOT_SAVE */
+
+/**
+ * Copy certain string literals into the given buffer in a specified format,
+ * which are valid identifiers and none of them are magic string.
+ *
+ * @return size of the literal-list in bytes, at most equal to the buffer size,
+ *         if the source parsed successfully and the list of the literals isn't empty,
+ *         0 - otherwise.
+ */
+size_t
+jerry_parse_and_save_literals (const jerry_char_t *source_p, /**< script source */
+                               size_t source_size, /**< script source size */
+                               bool is_strict, /**< strict mode */
+                               uint8_t *buffer_p, /**< buffer to save literals to */
```

Comment:
I think out_buffer_p or result_buffer_p would be better names (Only my personal preference)